File: cs50/lab1/pset1/credit.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    long input_number = get_long("Enter your credit card: ");
    long number = input_number;
    long odd_sum = 0;
    long even_sum = 0;
    int counter = 0;

// Loop checking sum correction
    while (number != 0)
    {
        long digit = number %  10;
        number = number / 10;
        counter++;
        if (counter % 2 == 1)
        {
            odd_sum = odd_sum + digit;
        }
        else
        {
            int even_digit = digit * 2;
            if (even_digit >= 10)
            {
                even_sum = even_sum + (even_digit % 10) + (even_digit / 10);
            }
            else
            {
                even_sum = even_sum + even_digit;
            }
        }
    }
    long card_correction = odd_sum + even_sum;

    long amex_test = pow(10, (counter - 2));
    long mastercard_test = pow(10, (counter - 2));
    long visa_test = pow(10, (counter - 1));
    //American Express
    if (card_correction % 10 == 0 && counter == 15 && ((input_number / amex_test) == 37 || input_number / amex_test == 34))
    {
        printf("AMEX\n");
    }
    //MasterCard
    else if (card_correction % 10 == 0 && counter == 16 && (input_number / mastercard_test == 51 || input_number / mastercard_test == 52 || input_number / mastercard_test == 53 || input_number / mastercard_test == 54 || input_number / mastercard_test == 55))
    {
        printf("MASTERCARD\n");
    }
    //Visa
    else if (card_correction % 10 == 0 && (counter == 13 || counter == 16) && input_number / visa_test == 4)
    {
        printf("VISA\n");
    }
    else
    {
        printf("INVALID\n");
    }
}

File: cs50/cs50_basics/lab6/sentimental-credit/credit.py
# TODO
while True:
    inputNumber = input('Card number: ')
    number = inputNumber
    if number.isnumeric() is True:
        if len(number) == 13 or len(number) == 15 or len(number) == 16:
            break

inputNumber = int(inputNumber)
number = int(number)

counter = 0
oddSum = 0
evenSum = 0

#Loop checking card correction
while (number != 0):
    digit = number % 10
    number = number // 10
    counter += 1
    if (counter % 2 == 1):
        oddSum = oddSum + digit
    else:
        evenDigit = digit * 2
        if evenDigit >= 10:
            evenSum = evenSum + (evenDigit % 10) + (evenDigit // 10)
        else:
            evenSum = evenSum + evenDigit

cardCorrection = oddSum + evenSum

amexTest = pow(10, (counter - 2))
mastercardTest = pow(10, (counter -2))
visaTest = pow(10, (counter - 1))

#American Express
if (cardCorrection % 10 == 0 and counter == 15 and (inputNumber // amexTest == 37 or inputNumber // amexTest == 34)):
    print('Amex')
elif (cardCorrection % 10 == 0 and counter == 16 and (inputNumber // mastercardTest == 51 or inputNumber // mastercardTest == 52 or inputNumber // mastercardTest == 53 or inputNumber // mastercardTest == 54 or inputNumber // mastercardTest == 55)):
    print('MASTERCARD')
elif (cardCorrection % 10 == 0 and (counter == 13 or counter == 16) and inputNumber // visaTest == 4):
    print('VISA')
else:
    print('INVALID')